src/codegen/compile.ml

```diff
@@ -11730,6 +11813,22 @@ and compile_prim_invocation (env : E.t) ae p es at =
   | OtherPrim "btstInt64", [_;_] ->
     const_sr (SR.UnboxedWord64 Type.Int64) (Word64.btst_kernel env)
 
+  (* Custom RTS functions *)
+  | OtherPrim s, _ when String.length s > 4 && String.sub s 0 4 = "rts:" ->
+    let s' = String.sub s 4 (String.length s - 4) in
+    if List.mem s' !(env.E.custom_rts_functions) then
+      const_sr SR.Vanilla (E.call_import env "rts" s')
+    else (
+      (* TODO: type checking error *)
+      Printf.printf "%s" Diag.(string_of_message {
+        sev = Error;
+        code = "M0199";
+        at;
+        cat = "RTS";
+        text = Printf.sprintf "custom function '%s' not found" s';
+      });
+      exit 1)
```

Comment:
As mentioned in the TODO, raising an error would be the nicer way. This would need some handling by the compiler pipeline.